components/ruby/lib/license_acceptance/strategy/file.rb

```diff
@@ -29,11 +29,23 @@ def accepted?(product_relationship)
 
         searching.each do |product|
           config.license_locations.each do |loc|
-            f = ::File.join(loc, product.filename)
-            if ::File.exist?(f)
-              logger.debug("Found license #{product.filename} at #{f}")
-              missing_licenses.delete(product)
-              break
+            license_path = ::File.join(loc, product.filename)
+            if ::File.exist?(license_path)
```

Comment:
You could avoid a level of nesting by returning early from the block.

```suggestion
            next unless ::File.exist?(license_path)
```